jsbits/Scenario.js

```diff
@@ -192,10 +194,22 @@ function gm$smartProcessFeatureCollection(fc, defVec, maxGeomId) {
     points.forEach(addGeomID);
     lines.forEach(addGeomID);
     surfaces.forEach(addGeomID);
+
+    // Get lat, lon, alt, srid if available
+    if (gi['lat'] && gi['lat'].constructor === Number &&
+        gi['lon'] && gi['lon'].constructor === Number &&
+        gi['alt'] && gi['alt'].constructor === Number) {
+        lat = gi['lat'];
+        lon = gi['lon'];
+        alt = gi['alt'];
+    }
+    if (gi['srid'] && gi['srid'].constructor === Number) {
+        srid = gi['srid']
+    }
+
     // transform everything from WGS84 to a metric reference system if needed
-    if(cmin.length >= 2 && cmax.length >= 2
-       && cmin[0] > -360 && cmax[0] < 360 && cmin[1] > -180 && cmax[1] < 180
-       && (cmax[0] - cmin[0]) < 5 && (cmax[1] - cmin[1]) < 5) {
```

Comment:
Might work.

Another option would be to check if 3rd coordinate is available (it is usually the case if we upload metric geometry). And then check the ratio height/(width+length) > xxx . If it is WGS'84, then it is super small, if it is metric the it the value is close to 1.
The only problem might arise if you really want upload a really flat 3D object :)

Maybe a combination of the two heuristics we outlined would be the best choice.